opm/parser/eclipse/EclipseState/EclipseState.hpp

```diff
@@ -123,7 +125,7 @@ namespace Opm {
 
 
         const UnitSystem& m_deckUnitSystem;
-        const ParseContext& m_parseContext;
+        const ParseContext m_parseContext;
```

Comment:
But since we're targeting a copy *anyway* it's most certainly not bad, and should allow some more optimisations because neither translation unit will have to care about aliasing.

Regarding your second point, the constructor now knows that its argument is a copy and will go out of scope with itself, so it should realise there's a move opportunity here, ensuring the actual copy is only performed once.